scipy/optimize/tests/test_least_squares.py

```diff
@@ -744,3 +746,30 @@ def test_basic():
     # test that 'method' arg is really optional
     res = least_squares(fun_trivial, 2.0)
     assert_allclose(res.x, 0, atol=1e-10)
+
+
+def test_small_tolerances_for_lm():
+    def res(p):
+        return y - H @ p
+
+    p0 = np.array([-0.62631426, 0.34736521, -1.53904034])
+    y = np.array([0.21042216, -0.774209, -1.69254418, -0.20248775, 2.15496739,
+                  0.57563308, 2.14036788, 1.9187491, -1.03636067, 3.58819572])
+    H = np.array([[0.4620464, -1.91682697, -0.55389804],
+                  [-0.79682045, -1.60787256, 1.58394725],
+                  [-0.52023376, -1.67694436, 0.18293149],
+                  [0.59203689, -1.37410441, -0.34793584],
+                  [-0.76610329, 0.39296886, -0.79426522],
+                  [-0.0211633, 1.20145896, 0.22471291],
+                  [-0.09212487, 0.35775539, -0.083304],
+                  [-0.91542477, 0.1543938, -1.14452311],
+                  [0.53072674, 1.28778141, 0.73426207],
+                  [-0.71480276, 0.66167675, -0.94704367]])
+
+    for ftol, xtol, gtol in [(0.0, 1e-13, 1e-13),
+                             (1e-13, 0.0, 1e-13),
+                             (1e-13, 1e-13, 0.0)]:
+        with warns(UserWarning): # UseWarning for EPS are shown.
+            print(ftol, xtol, gtol)
+            assert_raises(ValueError, least_squares, res, p0, xtol=xtol,
+                      ftol=ftol, gtol=gtol, method='lm')
```

Comment:
Aligning according to PEP8 will look better.

Reply:
I fixed it.